### PR TITLE
Releasing the 2020.4 SDK [Will merge only after 2020.4 ships]

### PR DESCRIPTION
**Pull request by jainam1995**
Will update the master branch with the changes https://github.com/tableau/connector-plugin-sdk/pull/658/files , after this MR is merged post the tableau 2020.4 release. Will also send out another MR for gh-pages merging and will post a blog post of the release updates in the end

### CODE REVIEW

File: tdvt/tdvt/version.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '2.2.1'
+__version__ = '2.1.17'
```
Comment (lukewrites) [inline tdvt/tdvt/version.py:1]: Make sure to pull `master` into your branch, we have some outdated code in it.

**Review by lukewrites — CHANGES_REQUESTED** [review]
Let's pull in `master` then take a look again, looks like some old code is in here.

File: tdvt/tdvt/version.py (hunk 1 above)
Comment (jainam1995) [reply tdvt/tdvt/version.py:1]: I actually just pulled in the master branch to 2020.4 : https://github.com/tableau/connector-plugin-sdk/pull/659/commits/5391e46e10a64ac83dce211cca89b0a3254c9826  . Guess i chose the wrong line during merge conflict resolution. Will update the line

**Comment by jainam1995** [comment]
> Let's pull in `master` then take a look again, looks like some old code is in here.
Pulled in the master branch to 2020.4 : https://github.com/tableau/connector-plugin-sdk/pull/659/commits/5391e46e10a64ac83dce211cca89b0a3254c9826

File: samples/scenarios/db_impersonation/connectionResolver.tdr [hunk 2]
```diff
@@ -6,6 +6,7 @@
     </connection-builder>
     <connection-normalizer>
       <required-attributes>
+        <setImpersonateAttributes/>
```
Comment (rosswbrown) [inline samples/scenarios/db_impersonation/connectionResolver.tdr:9]: I believe you first need to update dev-2020.4 to include the changes like [this](https://github.com/tableau/connector-plugin-sdk/pull/633) made directly against master, before merging dev-2020.4 into master.

Comment (jainam1995) [reply samples/scenarios/db_impersonation/connectionResolver.tdr]: Removed this line. As a part of https://github.com/tableau/connector-plugin-sdk/pull/659/commits/5391e46e10a64ac83dce211cca89b0a3254c9826 , have included the changes but had mistakenly manually kept this line

File: samples/scenarios/jdbc_kerberos/manifest.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?xml version='1.0' encoding='utf-8' ?>
+
+<connector-plugin class='postgres_jdbc_kerberos' superclass='jdbc' plugin-version='0.0.0' name='PostgreSQL JDBC with Kerberos' version='18.1' min-version-tableau='2020.3'>
```
Comment (pvanderknyff) [inline samples/scenarios/jdbc_kerberos/manifest.xml:3]:
```suggestion
<connector-plugin class='postgres_jdbc_kerberos' superclass='jdbc' plugin-version='0.0.0' name='PostgreSQL JDBC with Kerberos' version='18.1' min-version-tableau='2020.4'>
```
Double check that this can run in 2020.3, if not commit above change

File: tdvt/samples/connector-tests/connectionbuilder.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?xml version='1.0' encoding='utf-8' ?>
```
Comment (pvanderknyff) [inline tdvt/samples/connector-tests/connectionbuilder.xml:1]: Not part of this PR, but we should make sure these types of TDVT tests are documented.

File: tdvt/tdvt/tdvt_core.py [hunk 5]
```diff
@@ -25,11 +25,26 @@
 from .config_gen.test_config import TestSet
 from .resources import *
 from .tabquery import build_tabquery_command_line
+from .tabquery import build_connectors_test_tabquery_command_line
```
Comment (lukewrites) [inline tdvt/tdvt/tdvt_core.py:28]: this import can be on the previous line

File: tdvt/samples/connector-tests/connectionbuilder.xml (hunk 4 above)
Comment (jainam1995) [reply tdvt/samples/connector-tests/connectionbuilder.xml:1]: The documentation of this is already  a part of the gh pages 2020.4 branch . Will open a separate PR for for getting that into gh pages branch

File: tdvt/tdvt/tabquery.py [hunk 6]
```diff
@@ -51,6 +51,15 @@ def build_tabquery_command_line(work):
     cmdline = tb.build_tabquery_command_line(work)
     return cmdline
 
+def build_connectors_test_tabquery_command_line(conn_test_name, conn_test_file_name, conn_test_password_file):
```
Comment (lukewrites) [inline tdvt/tdvt/tabquery.py:54]: This function doesn't look for the env var at line 14. Can we use `configure_tabquery_path` & `build_tabquery_command_line` here?

File: tdvt/tdvt/tdvt_core.py [hunk 7]
```diff
@@ -25,11 +25,26 @@
...
+        self.conn_test_password_file = conn_test_password_file
+        self.timeout_seconds = 10
+
+    def run_connectors_test(self):
+        cmdline = build_connectors_test_tabquery_command_line(self.conn_test_name, self.conn_test_file, self.conn_test_password_file)
+        self.cmd_output = str(subprocess.check_output(cmdline, stderr=subprocess.STDOUT, universal_newlines=True,
+                                                      timeout=self.timeout_seconds))
+        print(self.cmd_output)
+        sys.exit(0)
+
```
Comment (lukewrites) [inline tdvt/tdvt/tdvt_core.py:47]: PEP violation needs another blank line

File: samples/scenarios/jdbc_kerberos/manifest.xml (hunk 3 above)
Comment (jainam1995) [reply samples/scenarios/jdbc_kerberos/manifest.xml]: Fixed. Confirmed with dsoni about this.

Comment (devanshsoni9) [reply samples/scenarios/jdbc_kerberos/manifest.xml]: This was tested when near branch was at 2020.4 so it is safe to change min version to 2020.4.

**Comment by jainam1995** [comment]
> Let's pull in `master` then take a look again, looks like some old code is in here.

Already done

File: CHANGELOG.md [hunk 8]
```diff
@@ -5,11 +5,16 @@
 ### Added
 - Add `vendor[1,2,3]-prompt` as options to ConnectionConfig-CT in `tcd_latest.xsd`.
 
+- Add JDBC Kerberos scenario for postgres 
+
 ### Changed
 - Validate length of `name` in Company-G in `connector_plugin_manifest_latest.xsd`
```
Comment (rosswbrown) [inline CHANGELOG.md:11]: This appears to be released as of 2020-09-21. [PR](https://github.com/tableau/connector-plugin-sdk/pull/605)

File: CHANGELOG.md [hunk 9]
```diff
@@ -5,11 +5,16 @@
 ### Added
 - Add `vendor[1,2,3]-prompt` as options to ConnectionConfig-CT in `tcd_latest.xsd`.
```
Comment (rosswbrown) [inline CHANGELOG.md:6]: Released, [PR](https://github.com/tableau/connector-plugin-sdk/pull/562)

File: CHANGELOG.md [hunk 10]
```diff
@@ -5,11 +5,16 @@
...
 
+- Add JDBC Kerberos scenario for postgres
```
Comment (rosswbrown) [inline CHANGELOG.md:8]: Also released

File: CHANGELOG.md (hunk 9 above)
Comment (jainam1995) [reply CHANGELOG.md:6]: Will remove this line

File: CHANGELOG.md (hunk 8 above)
Comment (jainam1995) [reply CHANGELOG.md:11]: Will remove this line

File: CHANGELOG.md (hunk 10 above)
Comment (jainam1995) [reply CHANGELOG.md:8]: This seems to be added in 2020.4 , [MasterBranch](https://github.com/tableau/connector-plugin-sdk/tree/master/samples/scenarios) vs [2020.4Branch](https://github.com/tableau/connector-plugin-sdk/tree/dev-2020.4/samples/scenarios). Seems i should keep this line ?

Comment (rosswbrown) [reply CHANGELOG.md:6]: Each of these that were previously in Unreleased should be moved to dated entries below.

Comment (jainam1995) [reply CHANGELOG.md:7]: Oh yes , fixed this and the previous two as well. Put them under the specified released date

**Review by rosswbrown — APPROVED** [review]

**Review by pvanderknyff — APPROVED** [review]
LGTM

**Review by lukewrites — CHANGES_REQUESTED** [review]
We need a minor version bump for TDVT along with all the other changes that go with that (changelog, readme)

Also can you see if your changes work when the tdvt path is specified with envvar rather than in a config file? Looking at the code, this may not work for run_file either

**Comment by jainam1995** [comment]
> We need a minor version bump for TDVT along with all the other changes that go with that (changelog, readme)
> 
> Also can you see if your changes work when the tdvt path is specified with envvar rather than in a config file? Looking at the code, this may not work for run_file either

Made the changes

**Review by lukewrites — APPROVED** [review]
🐑 🇮🇹
